runtime/src/qio/sys.c

```diff
@@ -547,12 +547,34 @@ err_t sys_lseek(fd_t fd, off_t offset, int whence, off_t* offset_out)
   return err_out;
 }
 
-err_t sys_stat(const char* path, struct stat* buf)
+
+void stat_to_sys_stat(const char* path, sys_stat_t* out_buf, struct stat* in_buf)
+{
+  stat(path, in_buf);
+  out_buf -> st_dev = in_buf -> st_dev;
```

Comment:
I don't think I've seen any code (in the Chapel codebase) that uses spaces around the `->` like this, so probably better to remove the spaces around the `->`.